src/USVPlugin.cpp

```diff
@@ -36,6 +39,10 @@ void USVPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
         boost::bind(&USVPlugin::updateBegin, this, _1)
     );
 
+    sdf::ElementPtr thrustersPluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "thrusters");
+    sdf::ElementPtr windPluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "wind_dynamics");
+    sdf::ElementPtr directForcePluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "direct_force");
```

Comment:
Do you really prefer making `dispatchPluginIfMatches` local instead of a static method?

Reply:
This is essentially a if-else if ...

```
if (plugin_name == "thrusters") {
   mThrusters = loadThrusters(thrustersPluginElement);
}
else if (plugin_name == "wind_dynamics") {
  mWind = loadWindParameters(windPluginElement);
}
else if (...) {
  ...
}
```

Reply:
I would prefer to avoid the if/else cascade, it's a problem keeping it as it is?

Reply:
what if

```c++
if (auto thrustersPluginElement = ::getPluginIfMatches(_plugin_sdf, "thrusters") {
   mThrusters = loadThrusters(thrustersPluginElement);
}
else if (auto windPluginElement = ::getPluginIfMatches(_plugin_sdf, "wind_dynamics") {
  mWind = loadWindParameters(windPluginElement);
}
else if (...) {
  ...
}

```

Reply:
but I think it is safer to check the plugin inside the `loadSomething` method, which is where we use them

Reply:
Maybe use switch-case? It's way cleaner then multiple if-else if clauses

Reply:
Also I am not sure, but I think defining a variable inside an if condition is only available after C++17

Reply:
I think what @doudou is suggesting is to do this

```
auto pluginName = plugin_sdf->Get<std::string>("name");
switch (pluginName)
{
    case "thrusters":
          mThrusters = loadThrusters(pluginName);
          break;
    case ....
}

```

Reply:
I am sorry, but I am going to override you guys here.

There is no way to find the current version clearer and more maintainable. Not even close. Like an order of magnitude.

The whole piece of code (~ 30 lines) is

~~~
auto pluginName = plugin_sdf->Get<std::string>("name");
if (pluginName == "thrusters") {
   mThrusters = loadThrusters(plugin_sdf);
}
else if (pluginName == "wind_dynamics") {
  mWind = loadWindParameters(plugin_sdf);
}
~~~

Try to forget what you know and read both pieces of code. Which one can you understand in less than 10 minutes ?

Reply:
fair point
```diff
-if (auto thrustersPluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "thrusters")) {
-    mThrusters = loadThrusters(thrustersPluginElement);
-    mRudders = loadRudders(thrustersPluginElement);
-}
-
-else if (auto windPluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "wind_dynamics")) {
-    mWind = loadWindParameters(windPluginElement);
-}
-
-else if (auto directForcePluginElement = ::dispatchPluginIfMatches(_plugin_sdf, "direct_force")) {
-    mDirectForce = loadDirectForceApplicationParameters(directForcePluginElement);
-}
+ auto pluginName = _plugin_sdf->Get<std::string>("name");
+ if ("thrusters" == pluginName) {
+     mThrusters = loadThrusters(_plugin_sdf);
+     mRudders = loadRudders(_plugin_sdf);
+ }
+ 
+ else if ("wind_dynamics" == pluginName) {
+     mWind = loadWindParameters(_plugin_sdf);
+ }
+ 
+ else if ("direct_force" == pluginName) {
+     mDirectForce = loadDirectForceApplicationParameters(_plugin_sdf);
+ }
```

Reply:
changed